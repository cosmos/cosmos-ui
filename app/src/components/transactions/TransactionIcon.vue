<template>
  <div class="tx__icon">
    <img :src="icon" :alt="transactionType" />
  </div>
</template>

<script>
import config from "src/../config"
export default {
  name: `transaction-icon`,
  props: {
    transactionType: {
      type: String,
      required: true,
    },
  },
  data: () => ({
    isExtension: config.isExtension,
  }),
  computed: {
    icon() {
      return this.isExtension
        ? `/icons/activity/${this.transactionType}.svg`
        : `/img/icons/activity/${this.transactionType}.svg`
    },
  },
}
</script>

<style>
.tx__icon {
  padding: 1rem;
}

.tx__icon img {
  max-height: 100%;
  max-width: 40px;
  display: block;
}

@media screen and (min-width: 1023px) {
  .tx__icon img {
    max-width: 48px;
  }
}
</style>
