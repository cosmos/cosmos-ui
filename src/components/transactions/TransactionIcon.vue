<template>
  <div class="tx__icon">
    <img :class="transactionGroup" :src="icon" :alt="transactionType" />
  </div>
</template>

<script>
export default {
  name: `transaction-icon`,
  props: {
    transactionGroup: {
      type: String,
      required: true
    },
    transactionType: {
      type: String,
      required: true
    }
  },
  computed: {
    icon() {
      if (this.transactionType.indexOf(` `) !== -1) {
        if (this.transactionType === `Update withdraw address`) {
          return `/img/icons/activity/Withdrawal.svg`
        } else {
          return `/img/icons/activity/${this.transactionType.substr(
            0,
            this.transactionType.indexOf(` `)
          )}.svg`
        }
      } else {
        return `/img/icons/activity/${this.transactionType}.svg`
      }
    }
  }
}
</script>

<style>
@import "../../styles/variables.css";

.tx__icon {
  padding: 1rem 0 1rem 1rem;
}

.tx__icon img {
  max-height: 100%;
  max-width: 52px;
  display: block;
}
</style>
