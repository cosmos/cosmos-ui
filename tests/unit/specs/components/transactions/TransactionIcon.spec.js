import { shallowMount } from "@vue/test-utils"
import TransactionIcon from "src/components/transactions/TransactionIcon"

describe(`TransactionIcon`, () => {
  let wrapper

  const transactions = [
    {
      type: `Sent`,
      group: `banking`,
      icon: `Sent`
    },
    {
      type: `Received`,
      group: `banking`,
      icon: `Received`
    },
    {
      type: `Staked`,
      group: `staking`,
      icon: `Staked`
    },
    {
      type: `Restaked`,
      group: `staking`,
      icon: `Restaked`
    },
    {
      type: `Unstaked`,
      group: `staking`,
      icon: `Unstaked`
    },
    {
      type: `Claimed`,
      group: `distribution`,
      icon: `Claimed`
    },
    {
      type: `Claimed with commission`,
      group: `distribution`,
      icon: `Claimed`
    },
    {
      type: `Update withdraw address`,
      group: `distribution`,
      icon: `Claimed`
    },
    {
      type: `Voted xxx`,
      group: `governance`,
      icon: `Voted`
    },
    {
      type: `Deposit`,
      group: `governance`,
      icon: `Deposit`
    },
    {
      type: `Submitted xxx proposal`,
      group: `governance`,
      icon: `Submitted`
    },
    {
      type: `Create`,
      group: `staking`,
      icon: `Create`
    },
    {
      type: `Edit validator`,
      group: `staking`,
      icon: `Edit`
    },
    {
      type: `Unjail`,
      group: `staking`,
      icon: `Unjail`
    }
  ]

  for (let i = 0; i < transactions.length; i++) {
    it(`renders a coloured ${transactions[i].type} transaction icon`, () => {
      wrapper = shallowMount(TransactionIcon, {
        propsData: {
          transactionGroup: transactions[i].group,
          transactionType: transactions[i].type
        }
      })
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.find(".tx__icon img").classes()).toContain(
        transactions[i].group
      )
      expect(wrapper.find(".tx__icon img").attributes("src")).toEqual(
        `/img/icons/activity/${transactions[i].icon}.svg`
      )
    })
  }
})
