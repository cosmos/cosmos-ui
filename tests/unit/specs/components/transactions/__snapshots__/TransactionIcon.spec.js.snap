// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionIcon renders a coloured Create transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Create"
    class="staking"
    src="/img/icons/activity/Create.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Deposit transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Deposit"
    class="governance"
    src="/img/icons/activity/Deposit.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Edit transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Edit"
    class="staking"
    src="/img/icons/activity/Edit.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Received transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Received"
    class="banking"
    src="/img/icons/activity/Received.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Restaked transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Restaked"
    class="staking"
    src="/img/icons/activity/Restaked.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Sent transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Sent"
    class="banking"
    src="/img/icons/activity/Sent.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Staked transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Staked"
    class="staking"
    src="/img/icons/activity/Staked.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Submitted transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Submitted"
    class="governance"
    src="/img/icons/activity/Submitted.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Unjail transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Unjail"
    class="staking"
    src="/img/icons/activity/Unjail.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Unstaked transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Unstaked"
    class="staking"
    src="/img/icons/activity/Unstaked.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Update withdraw address transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Update withdraw address"
    class="distribution"
    src="/img/icons/activity/Withdrawal.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Voted transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Voted"
    class="governance"
    src="/img/icons/activity/Voted.svg"
  />
</div>
`;

exports[`TransactionIcon renders a coloured Withdrawal transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Withdrawal"
    class="distribution"
    src="/img/icons/activity/Withdrawal.svg"
  />
</div>
`;
